lib/SILOptimizer/Mandatory/MoveOnlyAddressCheckerUtils.cpp

```diff
@@ -2210,17 +2235,9 @@ void MoveOnlyAddressCheckerPImpl::insertDestroysOnBoundary(
     if (!debugVar) {
       return;
     }
-    auto varInfo = debugVar.getVarInfo();
-    if (!varInfo) {
-      return;
-    }
-    SILBuilderWithScope debugInfoBuilder(insertPt);
-    debugInfoBuilder.setCurrentDebugScope(debugVar->getDebugScope());
-    debugInfoBuilder.createDebugValue(
-        debugVar->getLoc(),
-        SILUndef::get(debugVar.getOperandForDebugValueClone()->getType(),
-                      insertPt->getModule()),
-        *varInfo, false, true);
+    insertDebugValueAfter(insertPt, debugVar,
+      SILUndef::get(debugVar.getOperandForDebugValueClone()->getType(),
```

Comment:
Btw, for future reference. I introduced a new API on SILUndef that takes a SILValue and grabs the type/SILModule from it. https://github.com/apple/swift/blob/a1219cc6625991ac13b0187a801ac1eb358e92ee/include/swift/SIL/SILUndef.h#L36